atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -633,3 +494,250 @@ def run(self):
+            'release': release,
+            'source': "{0}#{1}".format(source.uri, source.commit_id),
+        })
+        return build
+
+    def get_server_dir(self):
+        """
+        Obtain koji_upload_dir value used for worker builds.
+        """
+        return get_koji_upload_dir(self.workflow)
+
+
+class KojiImportSourceContainerPlugin(KojiImportBase):
+
+    key = PLUGIN_KOJI_IMPORT_SOURCE_CONTAINER_PLUGIN_KEY  # type: ignore
+
+    def __init__(self, tasker, workflow,
+                 kojihub=None, url=None,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 blocksize=None,
+                 target=None, poll_interval=5):
+        super(KojiImportSourceContainerPlugin, self).__init__(tasker, workflow,
+                                                              kojihub, url,
+                                                              verify_ssl, use_auth,
+                                                              koji_ssl_certs, koji_proxy_user,
+                                                              koji_principal, koji_keytab,
+                                                              blocksize,
+                                                              target, poll_interval)
+
+    def get_output(self, worker_metadatas, buildroot_id):
+        registry = self.workflow.push_conf.docker_registries[0]
+
+        build_name = get_unique_images(self.workflow)[0]
+        pullspec = copy.deepcopy(build_name)
+        pullspec.registry = registry.uri
+
+        outputs, output_file = koji_get_output(workflow=self.workflow, buildroot_id=buildroot_id,
+                                               pullspec=pullspec, platform=os.uname()[4],
+                                               source_build=True, logs=None)
+
+        return outputs, output_file
+
+    def get_buildroot(self, worker_metadatas):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        buildroots = []
+
+        buildroot = koji_get_buildroot(build_id=self.build_id, tasker=self.tasker,
+                                       osbs=self.osbs, rpms=False)
+        buildroot['id'] = '{}-{}'.format(buildroot['container']['arch'], buildroot['id'])
+
+        registry = self.workflow.push_conf.docker_registries[0]
+        build_name = get_unique_images(self.workflow)[0].to_str()
+
+        manifest_digest = registry.digests[build_name]
+        digest_version = get_manifest_media_version(manifest_digest)
+        media_type = get_manifest_media_type(digest_version)
+
+        buildroot['extra']['osbs']['koji'] = {
+            'build_name': build_name,
+            'builder_image_id': {media_type: manifest_digest.default}
+        }
+
+        buildroots.append(buildroot)
+        return buildroots
```

Comment:
Looks this method only returns a list containing a single builtroot. Why not just `return [buildroot]` directly?

Reply:
I see. This was brought in from the general case where there might have been more than one...

Reply:
We have `buildroot_id = buildroot[0]['id']` in `combine_metadata_fragments()`, which is not differentiated by build type. Doesn't seem worthwhile to add complexity *there*, in order to (possibly) reduce confusion *here*.